### PR TITLE
feat: arrow `unique`, `is_duplicated`, `is_unique`

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #372 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

[Relevant arrow issue/conversation](https://github.com/apache/arrow/issues/30950)

### CODE REVIEW

File: narwhals/_arrow/dataframe.py [hunk 1]
```diff
@@ -387,3 +388,76 @@ def rename(self, mapping: dict[str, str]) -> Self:
...
+            df.append_column(col_token, pa.array(np.arange(len(self))))
+            .group_by(columns)
+            .aggregate([(col_token, "count")])
+        )
+        is_duplicated = pc.greater(
+            df.join(
+                row_count, keys=columns, right_keys=columns, join_type="inner"
+            ).column(f"{col_token}_count"),
+            1,
+        )
```
Comment (MarcoGorelli) [inline narwhals/_arrow/dataframe.py:412]: wow

**Review by MarcoGorelli — APPROVED** [review]
this is wild, thanks for doing this

**Comment by FBruzzesi** [comment]
Was this the fastest review ever?!

File: narwhals/_arrow/dataframe.py (hunk 1 above)
Comment (FBruzzesi) [reply narwhals/_arrow/dataframe.py:412]: Not _too_ proud of the join, but alternative was looping through value counts :|

**Pull request merged**